### PR TITLE
[Jetpack Social] Implement CTAs in post settings screen (part 1)

### PR DESCRIPTION
**Pull request by RenanLukas**
Part of #18571 

This PR partially implements the CTAs in post settings screen. It's a work in progress and there will be a part 2 where the CTAS are fully functional.

To test:
**Connection list and message container disabled for published post**
The switch widget status (enabled or disabled) for a published post is still not working, since we're not yet parsing the post metadata regarding the selected accounts to share that post.

1 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
2 - Select a site that has at least one social sharing connection;
3 - Open post settings of a published post: social sharing elements (connections and message container) should be disabled;

**Connection list and message container enabled for draft post**
Connection list switch tapping will be handled in my next PR.

1 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
2 - Select a site that has at least one social sharing connection;
3 - Open post settings of a draft post: social sharing elements (connections and message container) should be enabled;

**"Subscribe to share more" button**
1 - Log in using a .com account;
2 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
3 - Select a SH site that has at least one social sharing connection;
4 - Open post settings;
5 - Scroll to `"Social sharing"` section and tap the button `"Subscribe to share more"`: the WebView should be opened with the Jetpack Social Basic checkout page;

**Social sharing container gone on SH login**
1 - Log in using a SH site;
2 - Make sure `JetpackSocialFeatureConfig` feature flag is enabled;
3 - Select a site that has at least one social sharing connection;
4 - Open post settings: `"Social Sharing"` section should be gone.

**"Connect accounts" button**
1 - Remove all connected accounts in Social screen;
2 - Open posts list;
3 - Tap on a draft post;
4 - Tap the overflow menu and then "Post settings";
5 - Scroll to the "Social Sharing" section;
6 - Tap "Connect accounts" button;
7 - Select a social network to connect a new account;
8 - Tap "Connect" and "Go to web";
9 - Connect the social network account in the WebView;
10 - Move back to the post settings screen: the new connection should be shown in the "Social Sharing" section.

## Regression Notes
1. Potential unintended areas of impact
None (code behind feature flag)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing. Unit tests will be updated/implemented in another PR.

3. What automated tests I added (or what prevented me from doing so)
Unit tests will be updated/implemented in another PR.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt
Comment (RenanLukas) [inline WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt]: These tests will be updated in another PR.

File: WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModelTest.kt
Comment (RenanLukas) [inline WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModelTest.kt]: These tests will be updated in another PR.

File: WordPress/src/main/java/org/wordpress/android/util/extensions/CollectionExtensions.kt
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/util/extensions/CollectionExtensions.kt:5]: Just acknwoledging that by a quick check I was not able to find a Collection built-in method to check this....one would have expected that to be there, right? 😄

File: WordPress/src/main/java/org/wordpress/android/ui/posts/social/PostSocialSharingModelMapper.kt
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/social/PostSocialSharingModelMapper.kt]: Isn't it possible here that `shareLimit.sharesRemaining - connections.filter { it.isSharingEnabled }.size` becomes negative? Wdyt?

**Review by develric — COMMENTED** [review]
Hey @RenanLukas 👋 , thanks for the PR sir 🙇 ! Code LGTM and testing are mostly working as described. I have a couple of notes that I'm reporting below (plus an inline comment for your check in PostSocialSharingModelMapper):
- As we said in slack, we might want to use this PR as a feature branch and merge the chained PRs into this one we can use for final check the changes before to merge to trunk. (on a side note we might fix the commented out unit tests with a separate PR targeting this one)
- in this test step 5, I would expect the webview to open with the .com user authenticated, instead I get the login to .com screen (also this one we might investigate/fix in case in a separate PR targeting this one or trunk depending on when it is available)
>"Subscribe to share more" button
1 - Log in using a .com account;
2 - Make sure JetpackSocialFeatureConfig feature flag is enabled;
3 - Select a SH site that has at least one social sharing connection;
4 - Open post settings;
5 - Scroll to "Social sharing" section and tap the button "Subscribe to share more": the WebView should be opened with the Jetpack Social Basic checkout page;

Let me know wdyt 🙇

File: WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt
Comment (develric) [reply WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsJetpackSocialUiStateMapperTest.kt]: Fixed [here](https://github.com/wordpress-mobile/WordPress-Android/pull/18866)

File: WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModelTest.kt
Comment (develric) [reply WordPress/src/test/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModelTest.kt]: Fixed [here](https://github.com/wordpress-mobile/WordPress-Android/pull/18853)

File: WordPress/src/main/java/org/wordpress/android/util/extensions/CollectionExtensions.kt
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/util/extensions/CollectionExtensions.kt:5]: Definitely 😅

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialConnectionsContainer.kt
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialConnectionsContainer.kt]: 🛑  It seems that the Column was removed here and this Composable is emitting multiple Composables without an actual Container now, which is unexpected, as the behavior will depend on the parent Composable, so it's not deterministic anymore.

Wrap the content of `EditPostSettingsJetpackSocialConnectionsContainer` in a `Column` or force this component to be only called inside `ColumnScope` by changing the signature to:
```kotlin
@Composable
fun ColumnScope.EditPostSettingsJetpackSocialConnectionsContainer(...) {
    // ...
}
```

Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsJetpackSocialConnectionsContainer.kt]: Thanks, @thomashorta ! It's fixed.

**Review by develric — APPROVED** [review]
Thanks for the changes @RenanLukas 🙇 ! I also checked the webview and it now works without requiring the user to login 👍 ! Since unit tests are fixed now, also the feature is still disabled behind a feature flag and this is becoming a bit big of a PR, I think we might stop to use it as a feature branch and merge it to clean up things a bit for the chained PRs. I'm approving it and let you decide if merge it now 🙇 .

**Pull request merged**
